examples/px4-offboard-mode.cpp

```diff
@@ -181,26 +330,183 @@ static void set_mode_send(enum px4_modes mode)
     case PX4_MODE_OFFBOARD:
         cmd.param1 = PX4_MAIN_MODE_CUSTOM;
         cmd.param2 = PX4_CUSTOM_MODE_OFFBOARD;
-        target_mode_requested = mode;
         break;
-    case PX4_MODE_STABILIZED:
     default:
-        cmd.param1 = MAV_MODE_STABILIZE_ARMED;
-        target_mode_requested = PX4_MODE_STABILIZED;
-        break;
+        printf("Error: Mode not handled\n");
+        return;
     }
 
     mavlink_msg_command_long_encode(SYSTEM_ID, MAV_COMP_ID_ALL, &msg, &cmd);
     msg_send(&msg);
 }
 
+static void next_action_print(struct tasks *t)
+{
+    switch (t->action) {
+    case ARM_DISARM:
+        printf("Requesting %s...\n", t->param1 ? "arm" : "disarm");
+        break;
+    case SET_MODE:
+        printf("Requesting mode %s...\n", mode_names[PX4_MODE_OFFBOARD]);
+        break;
+    case MOVE_X:
+        printf("Moving %i meters in X....\n", t->param1);
+        break;
+    case MOVE_Y:
+        printf("Moving %i meters in Y....\n", t->param1);
+        break;
+    case MOVE_Z:
+        printf("Moving %i meters in Z....\n", t->param1);
+        break;
+    case WAIT_MOVE:
+        break;
+    case LAND:
+        printf("Requesting land...\n");
+        break;
+    case WAIT_LAND:
+        printf("Waiting landing...\n");
+        break;
+    case END:
+        printf("Offboard mission completed\n");
+        break;
+    default:
+        printf("Unknown action action id=%u\n", t->action);
+    }
+}
+
 static void timeout_callback()
 {
-    position_set_send(0, 0, 0, 0);
+    static float x = 0, y = 0, z = 0, yaw = 0;
+    static uint8_t task_list_index = 0;
```

Comment:
It do not happen because when the task list reach to END action it do not increment anymore but I added the code to check the index and the array size.